systemd/systemd.go

```diff
@@ -183,6 +186,20 @@ func Version() (int, error) {
 	return ver, nil
 }
 
+// EnsureAtLeast checks whether the installed version of systemd is greater or
+// equal than the given one. An error is returned if the required version is
+// not matched, and also if systemd is not installed or not working
+func EnsureAtLeast(requiredVersion int) error {
+	version, err := Version()
+	if err != nil {
+		return err
+	}
+	if version < requiredVersion {
+		return fmt.Errorf(`installed version %d is too old`, version)
```

Comment:
Should this be a separate error type? eg.
```
type systemdTooOldError struct {
    expected int
    got int
}

func (e *systemdTooOldError) Error() string {
    return fmt.Sprintf("systemd version %d is too old (expected at least %d)", e.got, e.expected)
}
```
Then here:
```
return &systemdTooOld{got: version, expected: requiredVersion}
```
then in quotas code you can just return:

```
return fmt.Errorf("cannot use quotas with incompatible systemd: %v", err)
```
which would produce eg. `cannot use quotas with incompatible systemd: systemd version 123 is too old (expected 345)`

Reply:
Nice suggestion, done!